unison-src/tests/console.u

```diff
@@ -29,24 +29,26 @@ namespace Console where
       -- this really should typecheck but doesn't for some reason
       -- error is that `simulate` doesn't check against `Request Console c -> r`,
       -- but seems like that `r` should get instantiated as `{State (..)} c`.
-      handle simulate in k (at 0 ins)
+      handle k (at 0 ins) with simulate
     {Console.write t -> k} ->
       io = State.get
       ins = fst io
       outs = snd io
       -- same deal here
-      handle simulate in k (State.set (ins, cons t outs))
+      handle k (State.set (ins, cons t outs)) with simulate
     {a} -> a
 
 (++) = (Text.++)
 
-x = handle Console.state ([],[]) in
-      handle Console.simulate in
+x = handle
+      handle
         use Console read write
         use Optional Some None
         write "What's your name?"
         case read of
           Some name -> write ("Hello" ++ name)
           None -> write "Fine, be that way."
+      with Console.simulate
+    with Console.state ([],[])
```

Comment:
Here's another example of stacked handlers. Doesn't seem quite as odd with indents.

Reply:
I wonder if this syntactic sugar would make sense?

```
  handle
    expression
  with
    handler1
    handler2
```

... probably not

Reply:
I [was thinking](https://unisonlanguage.slack.com/archives/CLKV43YE4/p1579212213100800?thread_ts=1579210507.097500&cid=CLKV43YE4) the maybe the block following `with` could be a term that either evaluates to a single handler function or a list of handler functions. So, e.g., `handle exp with [handler1, handler2]` would be allowed. But I guess that wouldn't just be a sugar if it requires evaluation.

Reply:
You can already handle multiple abilities in a single handler so I’d say no need for further special syntax (yet).

Reply:
@pchiusano Oh, I didn't realize you could do that! Cool!

Reply:
@pchiusano Do we have a way to compose handlers yet?  
`handle exp with (handler1 and handler2)`